### PR TITLE
Better spacing in text layer.

### PR DESCRIPTION
**Pull request by yurydelendik**
Alternative solution for #6019 (and fixes its PDF; attaching cleaned up version by @Snuffleupagus )

Improved /test/pdfs/pdf.pdf page 3 layout:
_screen shot 2015-11-02 at 8 43 45 am

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
Generally this is looking like a really nice improvement, but I happened to notice that in `ìssue3205r.pdf` this patch causes a slight regression in the textLayer:
https://www.dropbox.com/s/b7va9xuuodkrar3/issue3205r_spacing.png?dl=0

It might be an idea to add `issue3205r.pdf` as a `text` test as well!?

**Comment by yurydelendik** [comment]
> I happened to notice that in ìssue3205r.pdf this patch causes a slight regression in the textLayer

This is a really bad case for us -- it overprints text at the same spot: first is prints "log   log", then variable names, and after that operators. That makes it a really bad (I would say corrupted) PDF since it's not preserving a reading order. We can safely discard this a regression (it did not work properly in first place), and we probably have few issues already opened for this particular case.

File: test/test_manifest.json [hunk 1]
```diff
@@ -1290,6 +1290,13 @@
        "link": false,
        "type": "eq"
     },
+    {
+      "id": "issue6019",
+      "file": "pdfs/issue6019.pdf",
+      "md5": "7a2e5dda3b0fc5c2e9060e378a8cdc4e",
+      "rounds": 1,
+      "type": "eq"
```
Comment (Snuffleupagus) [inline test/test_manifest.json]: Shouldn't this be a `text` test instead?

**Comment by Snuffleupagus** [comment]
> We can safely discard this a regression (it did not work properly in first place), and we probably have few issues already opened for this particular case.

That seems fine by me, thanks for the explanation!

However, I'm also wondering about the test changes in `issue5972.pdf`, since they seem to regress issue #5972. Is that another case of a weird PDF file, or does this need to be addressed before we land this?

**Comment by yurydelendik** [comment]
Hmmm, that tough one, looking at the screenshot below I might say that #5995 was a regression ("after" means after #6588 patch):

_screen shot 2015-11-03 at 7 36 00 am

Looking at start of the letters, #5995 started moving all letters to the right due to artificial padding at the front of the text run. Per [spec](http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#page=252&zoom=125,-50,551): "When the glyph for each character in the string is rendered, Tc shall be added ..."

IMHO to fix #5972, is last char and word spacing is remembered to be used as a right padding of the div block in the text layer. What do you prefer: address it here, re-open #5972 or perhaps in #6590?

File: test/test_manifest.json (hunk 1 above)
Comment (yurydelendik) [inline test/test_manifest.json]: Yes, it is a text test. Thank you.

File: test/test_manifest.json [hunk 2]
```diff
@@ -1807,7 +1814,7 @@
        "link": false,
        "type": "load",
        "password": "\u00E6\u00F8\u00E5",
-       "about": "The password (���) is UTF8 encoded."
+       "about": "The password (���) is UTF8 encoded."
```
Comment (Snuffleupagus) [inline test/test_manifest.json]: Nit: there appears to be some kind of encoding issue here.

**Comment by Snuffleupagus** [comment]
> IMHO to fix #5972, is last char and word spacing is remembered to be used as a right padding of the div block in the text layer. What do you prefer: address it here, re-open #5972 or perhaps in #6590?

I've got no real preference; do whatever you think is the easiest :-)

If you choose to handle it in #6590, feel free to land the current PR with `r=me` (after fixing the encoding issue in `test_manifest.json`)., since other than that it looks *really* good.

**Comment by yurydelendik** [comment]
@Snuffleupagus thanks for review

**Pull request merged**